### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by rehanmohiuddin**
- Added Login
- Added Register
- Added Auth Context
- Made Snackbar Generic Component
- Made Generic Loader Component
- AuthUtility Objects After Success/Failure

### CODE REVIEW

File: src/Frontend/Context/Auth.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  loading: false,
+  error: "",
+  registration: {
+    processInitiated: false,
+    email: null,
+    isRegistered: false,
+  },
+  AuthMessage: null,
+};
+const LOGIN = "LOGIN";
```
Comment (spraveenofficial) [inline src/Frontend/Context/Auth.js]: You can move these Constants to the Other Folder.

File: src/Frontend/Context/Auth.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+const OPEN_LOGIN_COMP = "OPEN_LOGIN_COMP";
+const OPEN_REGISTER_COMP = "OPEN_REGISTER_COMP";
+const OPEN_AUTH_COMP = "OPEN_AUTH_COMP";
+
+const CLOSE_AUTH_COMP = "CLOSE_LOGIN_COMP";
+
+const AuthContext = createContext(loginObject);
+
+const AuthProvider = ({ children }) => {
+  const LoginReducer = (state = loginObject, action) => {
```
Comment (spraveenofficial) [inline src/Frontend/Context/Auth.js]: Same, this one too. You can move this to Reducers folder.

File: src/Frontend/Pages/Auth/index.css [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  margin-bottom: 16px;
+}
+
+.has-float-label input:focus {
+  border-bottom: 2px solid #2979ff;
+  transition: all 0.2s;
+}
+
+.has-float-label input:disabled {
+  border-bottom: 2px dotted #8a8a8a !important;
```
Comment (spraveenofficial) [inline src/Frontend/Pages/Auth/index.css]: Don't use !important. it's not good practice.

File: src/Frontend/Pages/Auth/index.css [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  pointer-events: none;
+}
+
+.has-float-label input:focus ~ .label {
+  color: #2979ff;
+}
+
+.helper {
+  display: block;
+  color: #6d6d6d;
```
Comment (spraveenofficial) [inline src/Frontend/Pages/Auth/index.css]: Instead of writing colors directly, you can make variables.

**Review by spraveenofficial — COMMENTED** [review]
Suggested, some changes.

File: src/Frontend/Context/Auth.js (hunk 1 above)
Comment (rehanmohiuddin) [reply src/Frontend/Context/Auth.js]: These will be at top of the file followed redux practices that particular reducer file constants will be at that particular file

File: src/Frontend/Pages/Auth/index.css (hunk 3 above)
Comment (rehanmohiuddin) [reply src/Frontend/Pages/Auth/index.css]: Yup noted will do that

**Pull request merged**